New docs content (#931)
liveblocks/liveblocks: docs/pages/concepts/how-liveblocks-works.mdx

---
meta:
  title: "How Liveblocks works"
  parentTitle: "Concepts"
  description:
    "Liveblocks is a real-time collaboration infrastructure for building
    performant collaborative experiences."
---

Liveblocks is a real-time collaboration infrastructure for building performant
collaborative experiences. Liveblocks is built around the following core
concepts: primitives, rooms, integrations, and platform.

## Primitives

Liveblocks is a fully integrated solution built around core primitives to embed
collaborative experiences: [Presence](/docs/primitives/presence),
[Broadcast](/docs/primitives/broadcast), and
[Storage](/docs/primitives/storage).

<Figure highlight>
  <Image
    src="/images/docs/concepts/primitives.png"
    alt="How Liveblocks works - Primitives"
    layout="responsive"
    width={768}
    height={440}
    quality={90}
  />
</Figure>

## Rooms

> A room is the digital space in which people collaborate. Primitives are
> associated with dedicated rooms.

Reviewer: ```suggestion
Rooms are digital spaces in which people collaborate. Primitives are
used within separate rooms.
```
I think this was too vague before. Better?

Author: I made some changes in between so this suggestion is outdated. I think what I have now is clearer, what do you think?

Thanks!

<Figure highlight>
  <Image
    src="/images/docs/concepts/rooms.png"
    alt="How Liveblocks works - Rooms"
    layout="responsive"
    width={768}
    height={440}
    quality={90}
  />
</Figure>

## Integrations

Liveblocks and its community provide integrations that developers can use to
integrate with the Liveblocks primitives.

<Figure highlight>
  <Image
    src="/images/docs/concepts/integrations.png"
    alt="How Liveblocks works - Integrations"
    layout="responsive"
    width={768}
    height={440}
    quality={90}
  />
</Figure>

## Platform

> Liveblocks provides a fully-hosted platform that gives powerful tools and a
> WebSocket edge infrastructure that scales to millions of users.

Reviewer: ```suggestion
Liveblocks provides a fully-hosted platform that enables powerful tools, and a
WebSocket edge infrastructure that scales to millions of users.
```
Optional, style
